FEEL power function using non integer exponent

SimonRinguette opened the pull request:
I have integrated the big-math library (https://github.com/eobermuhlner/big-math) licensed under MIT to add support for FEEL evaluation of non integer exponent in FEEL.

2 ** 3.5 would previously return 8 (same as 2 ** 3) but will now return the proper result.

This is going beyond the DMN 1.2 specification.

@etirelli You also mentioned that you used an aproximation for the sqrl() feel function, this library has a BigDecimal function for sqrt. If you want I have an additional commit that I could add to this PR to also change the sqrt FEEL function to use this new library.

I choose this library because of its license and very small footprint.

SimonRinguette commented:
@tarilabs As discussed, this is the PR to use the power as referenced in the TCK issue that I raised: https://github.com/dmn-tck/tck/issues/190

kiereleaseuser3 commented:
Can one of the admins verify this PR? Comment with 'ok to test' to start the build.

kiereleaseuser3 commented:
Can one of the admins verify this PR? Comment with 'ok to test' to start the build.

kiereleaseuser2 commented:
Can one of the admins verify this PR? Comment with 'ok to test' to start the build.

kiereleaseuser2 commented:
Can one of the admins verify this PR? Comment with 'ok to test' to start the build.

kiereleaseuser2 commented:
Can one of the admins verify this PR? Comment with 'ok to test' to start the build.

kiereleaseuser2 commented:
Can one of the admins verify this PR? Comment with 'ok to test' to start the build.

etirelli commented:
ok to test

etirelli commented:
@SimonRinguette thanks, I think we can use that library!

Also, since the library is being added, I think we should change sqrt() to use it. 

Finally, for this PR to be merged we need a unit test for it. Also, @tarilabs can you please help with adding the dependency to the top level pom file and checking with the prod guys if they can pull in the library and build in brew?

SimonRinguette commented:
@etirelli   I'm pushing the sqrt function and the two test cases. Do you want this in separate commit or should I rewrite this commit?

etirelli commented:
@SimonRinguette whatever is easier... separate commit on the same PR is fine, because we always squash when merging.. 

Thanks for this!

SimonRinguette commented:
@etirelli  @tarilabs I have rewritten my commit to include the sqrt() function implemented with the new library.

Also I realised that the compiled version of power (** operator) is implemented differently when using the compiling switch. I have also contributed the compiled code for the ** operator.

I have added a new ** test case with non integer exponent and also two new test cases with sqrt().

Let me know if I need to update something in the PR for it to be acceptable for merge.

etirelli commented:
@SimonRinguette thanks! tomorrow @tarilabs can review and merge the code. The only other thing we need is to move the dependency version into the pom file, but we can do that on our side.

etirelli commented:
ok to test

tarilabs commented on kie-dmn/kie-dmn-feel/pom.xml:
```diff
@@ -46,6 +46,13 @@
+    <dependency>
+        <groupId>ch.obermuhlner</groupId>
+        <artifactId>big-math</artifactId>
+        <version>2.0.1</version>
```
I will move version on this side.

mbiarnes replied:
If this have to be added please add this to the kie-parent:
in properties <version.ch.obermuhlner>2.0.1</version>
and in dependencyManagement
<!-- a comment why this is needed -->
<dependency>
  <groupId>ch.obermuhlner</groupId>
  <artifactId>big-math</artifactId>
</dependency>
and then please remove the <version>2.0.1</version> from here.

tarilabs commented:
CI green, downstream compilation failed for reasons unrelated to this PR

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.7.0:testCompile (default-testCompile) on project drools-wb-guided-dtable-editor-client: Compilation failure
12:29:38 [ERROR] /home/jenkins/workspace/KIE/7.11.x/pullrequest/drools-compile-downstream-build/downstream-repos/drools-wb/drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/table/columns/ConsumerFactoryTest.java:[183,24] incompatible types: org.uberfire.ext.wires.core.grids.client.model.GridCellValue is not a functional interface
12:29:38 [ERROR]     multiple non-overriding abstract methods found in interface org.uberfire.ext.wires.core.grids.client.model.GridCellValue
```

Will ask for kindly merge please.

tarilabs commented:
Thanks a lot @SimonRinguette !
